source/common/config/delta_subscription_state.cc

```diff
@@ -137,10 +137,32 @@ DeltaSubscriptionState::getNextRequestAckless() {
     }
     names_removed_.clear();
   }
-  std::copy(names_added_.begin(), names_added_.end(),
+  std::vector<std::string> resource_names_added;
+  std::vector<std::string> resource_names_removed;
+  std::vector<udpa::core::v1::ResourceLocator> resource_urls_added;
+  std::vector<udpa::core::v1::ResourceLocator> resource_urls_removed;
+  for (std::string const& resource : names_added_) {
+    if (resource.substr(0, 7) == "udpa://") {
+      resource_urls_added.push_back(UdpaResourceIdentifier::decodeUrl(resource));
+    } else {
+      resource_names_added.push_back(resource);
+    }
+  }
+  for (std::string const& resource : names_removed_) {
+    if (resource.substr(0, 7) == "udpa://") {
+      resource_urls_removed.push_back(UdpaResourceIdentifier::decodeUrl(resource));
+    } else {
+      resource_names_removed.push_back(resource);
+    }
```

Comment:
FYI, while this string comparison business may seems kind of inefficient, it actually allows folks to use udpa:// strings directly as a resource name without having to populate the proto in configs. For bootstrap it might not matter, but it might help control planes incrementally adopt URLs.